### PR TITLE
Add core-dns image by digest for CNAO

### PR DESCRIPTION
**Pull request by oshoval**
Signed-off-by: Or Shoval <oshoval@redhat.com>

**Reviewer Checklist**
- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:
```release-note
None
```

### CODE REVIEW

**Comment by oshoval** [comment]
/cc @orenc1 @tiraboschi 

Followed https://github.com/kubevirt/hyperconverged-cluster-operator/pull/1791
didn't test it locally, is there something that need to be run or the CI lanes are enough ?

**Comment by oshoval** [comment]
We dont have a tag like 4.11 for core-dns so it faiis
We can change for a version tag, but considering it again with Simone, we can just let CNAO manage it
in case KubeSecondaryDNS will update the image, CNAO will bump the changed, and once there is a release of CNAO
it will updated HCO's CSV

Closing

File: deploy/images.env [hunk 1]
```diff
@@ -54,3 +55,4 @@ DIGEST_LIST="${DIGEST_LIST},k8s.gcr.io/sig-storage/csi-snapshotter@sha256:818f35
 DIGEST_LIST="${DIGEST_LIST},k8s.gcr.io/sig-storage/csi-provisioner@sha256:4e74c0492bceddc598de1c90cc5bc14dcda94cb49fa9c5bad9d117c4834b5e08"
 DIGEST_LIST="${DIGEST_LIST},quay.io/openshift/origin-kube-rbac-proxy@sha256:baedb268ac66456018fb30af395bb3d69af5fff3252ff5d549f0231b1ebb6901"
 DIGEST_LIST="${DIGEST_LIST},quay.io/kubevirt-ui/kubevirt-plugin@sha256:253cb213f052faec5b41e56860133ebb7508715f6abe691b2609c9e36dec62ab"
+DIGEST_LIST="${DIGEST_LIST},k8s.gcr.io/coredns/coredns@sha256:5b6ec0d6de9baaf3e92d0f66cd96a25b9edbce8716f5f15dcd1a616b3abd590e"
```
Comment (oshoval) [inline deploy/images.env:58]: @tiraboschi 
If we close this PR, we won't have this change on deploy/images.env
so maybe we do want it ?

**Comment by oshoval** [comment]
Fixed CORE_DNS_VERSION
reopened for discussion of https://github.com/kubevirt/hyperconverged-cluster-operator/pull/2190#discussion_r1054457838

File: deploy/images.env (hunk 1 above)
Comment (tiraboschi) [reply deploy/images.env:58]: Yes, correct but do we need it?
The question is what is going to keep `CORE_DNS_VERSION="v1.8.6"` up to date with a properly tested version.

Comment (oshoval) [reply deploy/images.env:58]: there isn't automation for that atm
note that for kube-rbac there isn't as well 
https://github.com/kubevirt/hyperconverged-cluster-operator/pull/1791#discussion_r815645260

Is it better from nothing ? if we don't have this PR, it means we can't deploy it offline with KSD, 
is it fine or a must ?

Comment (tiraboschi) [reply deploy/images.env:58]: offline deployment is a must,
I was supposing it was already handled con CNAO side, if it's not this is better than nothing

Comment (oshoval) [reply deploy/images.env:58]: so keeping the PR open thanks

Comment (oshoval) [reply deploy/images.env:58]: Note that i don't see in this list the kubemacpool image for example
so maybe this list need just the stuff that is deployed as part of CNAO and not the images
that belong to the components of CNAO ?

Comment (oshoval) [reply deploy/images.env:58]: You are right, it is CNAO that updates the relatedImages in CSV 
and then it is aggregated on HCO
so we can close the PR
HCO is responsible only to what it deploys and gets from CNAO, what CNAO deploys

We can see here it is on the csv of HCO
https://github.com/kubevirt/hyperconverged-cluster-operator/blob/a02b1f26451d1cff1d647942caf3b9135aa58af4/deploy/index-image/community-kubevirt-hyperconverged/1.9.0/manifests/kubevirt-hyperconverged-operator.v1.9.0.clusterserviceversion.yaml#L3724

**Comment by oshoval** [comment]
See https://github.com/kubevirt/hyperconverged-cluster-operator/pull/2190#discussion_r1057226551

**Pull request closed without merging**